pkg/sources/postgres/dogs.go

```diff
@@ -53,7 +53,7 @@ func (s *Store) CreateDog(ctx context.Context, dog *models.Dog) (*models.Dog, er
 	if err != nil {
 		return nil, err
 	}
-	return dog, nil
```

Comment:
The problem here is that we're returning the _memory_ copy of the `Dog`, which is the exact same memory location as what was passed in. This does not then do a round-trip retrieval from the DB, meaning that we don't verify that what comes out of the DB actually agrees with what got put in.
For example, if there was a new field added to `Dog`, but we forgot to update the SQL to write that to the DB, and there was a test ensuring the input & output of this method were field-by-field matching, that would show a false positive.